patches/chrome-browser-profiles-profile_impl.cc.patch

```diff
@@ -10,12 +10,13 @@ index 3c9591c492dc11b9ab9eaae4e9f2a073dc822107..5b2d3dfaee1d1b8f09367848571d7b92
  #include "build/build_config.h"
  #include "chrome/browser/background/background_contents_service_factory.h"
  #include "chrome/browser/background_fetch/background_fetch_delegate_factory.h"
-@@ -529,6 +530,9 @@ ProfileImpl::ProfileImpl(
+@@ -529,6 +530,10 @@ ProfileImpl::ProfileImpl(
    BrowserContextDependencyManager::GetInstance()
        ->RegisterProfilePrefsForServices(this, pref_registry_.get());
  
 +  // Done here instead of browser_prefs.cc so it can override default prefs
 +  brave::RegisterProfilePrefs(pref_registry_.get());
++  brave::OverrideDefaultProfilePrefs(pref_registry_.get());
```

Comment:
I don't think we should add to this patch. If we need a method that doesn't override the default pref values then we should create something like `brave::RegisterProfilePrefsWithoutDefaultOverrides` and use that in the testing profile. The original brave::RegisterProfilesPrefs would call that method and also call `OverrideDefaultProfilePrefs`

Reply:
Hmm, why do you think so? Now method names 100% reflect what they do. If a method does both registering and overriding, it should be named `RegisterProfilePrefsAndOverrideDefaults`.

Reply:
I don't care which way it goes as long as we only patch with a single method call here

Reply:
actually there's a better change for this. We do a browser_prefs.cc override and always calls `brave::RegisterProfilePrefs` inside `RegisterProfilePrefs` and then change this patch to only call `brave::OverrideDefaultProfilePrefs`. Then the test_profile.cc patch can be removed completely

Reply:
going this route we wouldn't change the method names

Reply:
Looks good, will try this!

Reply:
you can DM me for more info on chromium_src overrides